### PR TITLE
cannon: Enable 64-bit tests

### LINKED ISSUE #12252: Enable 64-bit MT-Cannon Go tests in CI
Run 32- and 64-bit MT-Cannon tests in CI. See https://github.com/ethereum-optimism/optimism/blob/a12738b7b17f5b2cc00d45c0545866f0b8cae5d4/.circleci/config.yml#L202C1-L212C42

### PR DESCRIPTION
**Pull request by mbaxter**
**Description**

Enable 64-bit Cannon tests in CI.  

Updates a handful of tests but most are deferred for future PRs (skipped).  Skipped tests will be enabled in follow-up PRs.  See follow-up issue: https://github.com/ethereum-optimism/optimism/issues/12598

**Metadata**

Fixes https://github.com/ethereum-optimism/optimism/issues/12252

### CODE REVIEW

File: cannon/Makefile [hunk 1]
```diff
@@ -86,23 +86,33 @@ cannon-stf-verify:
...
 	go test $(FUZZLDFLAGS) -run NOTAREALTEST -v -fuzztime 10s -fuzz=FuzzStateSyscallCloneST ./mipsevm/tests
-	# Multi-threaded tests
-	go test $(FUZZLDFLAGS) -run NOTAREALTEST -v -fuzztime 10s -fuzz=FuzzStateSyscallCloneMT ./mipsevm/tests
+	# Multi-threaded 32-bit tests
+	go test $(FUZZLDFLAGS) -run NOTAREALTEST -v -fuzztime 10s -fuzz=FuzzStateSyscallCloneMT32 ./mipsevm/tests
 	# 64-bit tests - increased fuzztime for a larger input space
 	go test $(FUZZLDFLAGS) -tags=cannon64 -run NOTAREALTEST -v -fuzztime 20s -fuzz=FuzzStateConsistencyMulOp ./mipsevm/tests
 	go test $(FUZZLDFLAGS) -tags=cannon64 -run NOTAREALTEST -v -fuzztime 20s -fuzz=FuzzStateConsistencyMultOp ./mipsevm/tests
 	go test $(FUZZLDFLAGS) -tags=cannon64 -run NOTAREALTEST -v -fuzztime 20s -fuzz=FuzzStateConsistencyMultuOp ./mipsevm/tests
+	go test $(FUZZLDFLAGS) --tags=cannon64 -run NOTAREALTEST -v -fuzztime 20s -fuzz=FuzzStateSyscallBrk64 ./mipsevm/tests
```
Comment (mbaxter) [inline cannon/Makefile]: I created fuzz tests for 32- and 64-bit vms that are functionally the same but just named differently.  This avoids errors about the corpus data being the wrong type (uint32 vs uint64 Words).

File: cannon/mipsevm/tests/evm_common_test.go [hunk 2]
```diff
@@ -47,10 +47,7 @@ func TestEVM(t *testing.T) {
 				// Short-circuit early for exit_group.bin
 				exitGroup := f.Name() == "exit_group.bin"
 				expectPanic := strings.HasSuffix(f.Name(), "panic.bin")
-
-				evm := testutil.NewMIPSEVM(c.Contracts)
-				evm.SetLocalOracle(oracle)
-				testutil.LogStepFailureAtCleanup(t, evm)
+				validator := testutil.NewEvmValidator(t, c.StateHashFn, c.Contracts, testutil.WithLocalOracle(oracle))
```
Comment (mbaxter) [inline cannon/mipsevm/tests/evm_common_test.go:50]: Cleaned up some of the EVM validation logic so we always do this validation through `testutil` helpers.   Creating this validator object avoids calling `LogStepFailureAtCleanup` at every step, which would be problematic.

File: cannon/Makefile (hunk 1 above)
Comment (Inphi) [reply cannon/Makefile]: You might need to merge from develop to fix a conflict here. Develop contains new 64-bit fuzz tests.

File: .circleci/config.yml [hunk 3]
```diff
@@ -162,9 +162,6 @@ jobs:
         description: Whether to notify on failure
         type: boolean
         default: false
-      mips64:
-        type: boolean
-        default: false
```
Comment (Inphi) [inline .circleci/config.yml:167]: This was setup so we could run cannon 32 and 64 tests in a matrix. Which speeds up the `main` CI workflow as the `cannon-go-lint-and-test` job can run in parallel. Looking at the [latest build times](https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/70454/workflows/871bf6e5-b967-4791-a201-05c28b889443/jobs/2905451), it takes 7 minutes to run both jobs from a cached Go build, and half of that time is evenly split between 32 and 64 bit test runs.

So I suggest keeping the matrix setup to reduce build times.

**Review by Inphi — COMMENTED** [review]
The test changes look good. However, I think the 64-bit tests should run in a matrix alongside 32-bits. Lmkwyk.

File: cannon/Makefile (hunk 1 above)
Comment (Inphi) [reply cannon/Makefile]: nvmind my comment. I see the change is already in there.

File: .circleci/config.yml (hunk 3 above)
Comment (mbaxter) [reply .circleci/config.yml:167]: Yeah - that makes total sense.  I just don't know what I'm doing with circleci 😂  I will fiddle with it.

**Review by Inphi — APPROVED** [review]
LGTM! Also kicked off the recurring expensive fp tests. We should merge once those pass.

**Comment by Inphi** [comment]
@mbaxter Incoming merge conflict with https://github.com/ethereum-optimism/optimism/pull/12765

**Comment by mslipper** [comment]
@mbaxter resolved conflicts and added to queue. The old branch restriction has been removed.

**Pull request merged**